sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2077,9 +2077,11 @@ object SQLConf {
 
   val LEGACY_CTE_PRECEDENCE_ENABLED = buildConf("spark.sql.legacy.ctePrecedence.enabled")
     .internal()
-    .doc("When true, outer CTE definitions takes precedence over inner definitions.")
+    .doc("When true, outer CTE definitions takes precedence over inner definitions. If set to " +
+      "false, inner CTE definitions take precedence. The default value is empty, which keep the " +
```

Comment:
Actually, `keep the behavior` is not correct because the 2.4.x behavior is `true`. The default value is new behavior in 3.0, isn't it?

Reply:
Yes, use `keep` here is ambiguous, thanks for pointing out! I want to express `always`.

Reply:
I'm not sure how exact and detailed this description needs to be, but
- if it is set to `false` then an inner CTE always takes precedence,
- if it is set to `true` then it complies with the (bit weird) 2.4 behaviour, where an outer CTE takes precedence in almost all of the cases (ie. inner CTE is in another CTE definition or in a subquery expression), except for the case where inner CTE is in a subquery.